src/textual/widgets/_data_table.py

```diff
@@ -655,6 +657,84 @@ def get_cell_at(self, coordinate: Coordinate) -> CellType:
         row_key, column_key = self.coordinate_to_cell_key(coordinate)
         return self.get_cell(row_key, column_key)
 
+    def get_row(self, row_key: RowKey | str) -> list[CellType]:
+        """Get the values from the row identified by the given row key.
+
+        Args:
+            row_key: The key of the row.
+
+        Returns:
+            A list of the values contained within the row.
+
+        Raises:
+            RowDoesNotExist: When there is no row corresponding to the key.
+        """
+        if row_key not in self._row_locations:
+            raise RowDoesNotExist(f"Row key {row_key!r} is not valid.")
+        cell_mapping: dict[ColumnKey, CellType] = self._data.get(row_key, {})
+        ordered_row: list[CellType] = []
+        for column in self.ordered_columns:
```

Comment:
Could we do this in a list expression? Highly unlikely there will be enough columns to make a difference, but you never know.